plugin/trino-opa/src/main/java/io/trino/plugin/opa/OpaConfig.java

```diff
@@ -28,6 +28,8 @@ public class OpaConfig
     private boolean logRequests;
     private boolean logResponses;
     private boolean allowPermissionManagementOperations;
+    private Optional<URI> opaRowFiltersUri = Optional.empty();
+    private Optional<URI> opaColumnMaskingUri = Optional.empty();
```

Comment:
This is a great idea, we were playing with a similar concept. The one thing I'm unsure about is allowing a configuration endpoint to determine whether row filtering is applied at all, for instance - as that feels more like a configuration responsibility. Kind of like if the OAuth config endpoint allowed you to disable auth altogether.

Not opposed to it, just some thoughts.
______

**An alternative:**

What's interesting is that we don't technically need these to be different URIs, they could just be different keys from the response and we could do everything OPA-side without including another type of request.

Suppose we had:
- `opa.policy.uri = https://opa/v1/data/my_policy/allow`
- `opa.policy.batched-uri = https://opa/v1/data/my_policy/batchAllow`
- `opa.policy.row-filters-uri = https://opa/v1/data/my_policy/rowFilters`
- `opa.policy.column-masking-uri = https://opa/v1/data/my_policy/columnMask`

This is equivalent to querying `https://opa/v1/data/my_policy` for all possible types of request and pulling the relevant field from the response (`allow`, `batchAllow`, `rowFilters` or `columnMask`).

The OPA policy itself can even directly determine what result to provide so that Trino doesn't need to pop fields from the response manually. This is because the request unequivocally determines what Trino is looking for:
- If `operation` is `GetRowFilters`: the relevant result is the row filters decision
- If `operation` is `GetColumnMask`: the relevant result is the column mask decision
- If `action.filterResources` is present: the relevant result is the batch filtering decision
- Otherwise, the relevant result is the standard boolean `allow` decision

Some care would need to be exercised for performance (but we can provide a small rego wrapper to deal with this)

The config would then become:
- `opa.policy.unified-uri` (or whatever we call this new "catch-all" URI)
- `opa.policy.enable-row-filtering`
- `opa.policy.enable-column-masking`

The last two would allow us to disable row filtering and column masking from the config in case there's performance concerns

Maybe we could take this into an issue? CC  @sbernauer